packages/forma-36-react-components/src/components/Asset/AssetIcon/AssetIcon.tsx

```diff
@@ -14,32 +14,26 @@ export interface AssetIconProps
   type?: AssetType;
 }
 
-const defaultProps: Partial<AssetIconProps> = {
-  type: 'archive',
-  testId: 'cf-ui-asset-icon',
-};
-
 /**
  * Renders only the Illustration that would represent this asset's type
  */
-export class AssetIcon extends Component<AssetIconProps> {
-  static defaultProps = defaultProps;
-
-  render() {
-    const { type, ...otherProps } = this.props;
-
-    let illustrationName = type!.charAt(0).toUpperCase() + type!.slice(1); // eslint-disable-line @typescript-eslint/no-non-null-assertion
-    if (!isIllustrationType(illustrationName)) {
-      illustrationName = DEFAULT_ILLUSTRATION_NAME;
-    }
-
-    return (
-      <Illustration
-        illustration={illustrationName as IllustrationType}
-        {...otherProps}
-      />
-    );
+export function AssetIcon({ type, ...otherProps }: AssetIconProps) {
+  let illustrationName = type!.charAt(0).toUpperCase() + type!.slice(1); // eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
Why do we need to disable @typescript-eslint/no-non-null-assertion?

I'm actually not very familiar with  non-null assertions, but can't we refactor it instead of disabling the rule?

Something from these example, maybe?
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-non-null-assertion.md
https://palantir.github.io/tslint/rules/no-non-null-assertion/

Reply:
The problem is the default props are used currently. IMO this would be solved by moving from `defaultProps` to "normal" JS with default values. So we go from 

```jsx
export function Asset({
  className,
  src,
  status,
  testId,
  title,
  type,
  ...otherProps
}: AssetProps): React.ReactElement { ... }

Asset.defaultProps = {
  type: 'image',
  testId: 'cf-ui-asset',
};
```

to

```jsx
export function Asset({
  className,
  src,
  status,
  testId = 'cf-ui-asset',
  title,
  type = 'image',
  ...otherProps
}: AssetProps): React.ReactElement { ... }
```

Reply:
Ohhhhh I prefer that too, but a while ago we had this proposal
https://github.com/contentful/forma-36/pull/574

And because of that, it is done the way it is now

Reply:
I don't remember exactly why we could not use the defaults in the arguments, but maybe the proposal has some explanation

Reply:
maybe @andipaetzold has a better memory

Reply:
Yeah I remember seeing that. I think it can be kind of a first step towards what I mentioned, since the default values are a pretty big change that only work with function components. That's why I'm working on refactoring all our classes to functions first, then after that we can look into changing from `defaultProps` to default values if we want to 🙂 Then once that is done we can get rid of non-null assertions.

Reply:
Yes, using the "normal" function components default values instead of `defaultProps` should solve this issue and the types would be correct. Especially as ["defaultProps on functions will eventually get deprecated."](https://twitter.com/dan_abramov/status/1133878326358171650), I think we should try to slowly get away from `defaultProps`.

In the mentioned PR I didn't get rid of `defaultProps` because I would have to convert all class components to functional components and wanted to do have a simple and consistent migration. If all components were functional components, I would have dropped `defaultProps`.

I have something in my head that the storybook might break if we remove the `defaultProps`, but I do not fully remember.